frontend/src/app/legend-card/legend-card.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { MatSlideToggleChange } from '@angular/material/slide-toggle';
+import { Record, RecordsOneChannel } from '../chart/record';
+
+@Component({
+  selector: 'legend-card',
+  templateUrl: './legend-card.component.html',
+  styleUrls: ['./legend-card.component.scss'],
+})
+export class LegendCardComponent implements OnInit {
+  @Input() time: string;
+  @Input() date: string;
+  @Input() power: number;
+
+  @Input() recordsOneChannel: RecordsOneChannel;
+  @Output() showChange = new EventEmitter<[string, boolean]>();
+
+  checked = true;
+
+  constructor() {}
+
+  getAvg() {
+    if (this.recordsOneChannel?.data.length == 0) return 0;
+    let sum = 0;
+    this.recordsOneChannel.data.forEach((record: Record) => {
```

Comment:
Avoid using foreach in ts.
Below is the quote from the typescript style guide.
Do not use Array.prototype.forEach. It makes code harder to debug and defeats some useful compiler checks (e.g. reachability).

someArr.forEach((item, index) => {
  someFn(item, index);
});
 Why?
Consider this code:

let x: string|null = 'abc';
myArray.forEach(() => { x.charAt(0); });
You can recognize that this code is fine: x isn't null and it doesn't change before it is accessed. But the compiler cannot know that this .forEach() call doesn't hang on to the closure that was passed in and call it at some later point, maybe after x was set to null, so it flags this code as an error. The equivalent for-of loop is fine.

See the error and non-error in the playground.

In practice, variations of this limitation of control flow analysis show up in more complex codepaths where it is more surprising.